tldw_Server_API/app/core/Evaluations/circuit_breaker.py

```
@@ -87,14 +95,32 @@ async def call(self, func: Callable, *args, **kwargs) -> Any:
+        # In CLOSED state, acquire a permit to bound concurrent in-flight calls.
+        closed_permit_acquired = False
+        if state_snapshot == CircuitState.CLOSED:
+            await self._closed_semaphore.acquire()
+            closed_permit_acquired = True
+            # After acquiring, re-check if circuit opened while we waited.
+            async with self._lock:
+                if self.state == CircuitState.OPEN:
+                    self.stats.rejected_calls += 1
+                    if closed_permit_acquired:
+                        self._closed_semaphore.release()
+                    raise CircuitOpenError(f"Circuit breaker {self.name} is OPEN")
```

Ruff (0.14.3)

Avoid specifying long messages outside the exception class

(TRY003)